src/main/java/org/beanpod/switchboard/controller/DecoderController.java

```diff
@@ -29,24 +32,38 @@
 @RequestMapping("/decoder")
 @RequiredArgsConstructor
 public class DecoderController {
+
   public static final String UNKNOWN_ERROR_MESSAGE = "Unknown error in DecoderController";
 
   private final DecoderDaoImpl decoderDao;
   private final DeviceDaoImpl deviceService;
   private final DecoderMapper decoderMapper;
   private final StreamMapper streamMapper;
   private final DecoderService decoderService;
+  private final MaintainDeviceStatus maintainDeviceStatus;
 
   @GetMapping
   public List<DecoderDto> retrieveAllDecoders() {
     List<DecoderEntity> decoderEntity = decoderDao.getDecoders();
+    maintainDeviceStatus.maintainStatusField(decoderEntity);
     return decoderMapper.toDecoderDtos(decoderEntity);
   }
 
   @GetMapping("/{serialNumber}")
   public ResponseEntity<DecoderDto> retrieveDecoder(@PathVariable @Valid String serialNumber) {
-    return decoderDao
-        .findDecoder(serialNumber)
+    // maintain status field and create a log if status changed
+    Optional<DecoderDto> decoder = decoderDao.findDecoder(serialNumber);
+    if (decoder.isPresent()) {
+      List<DecoderEntity> decodersListTemp = new LinkedList<>();
+      decodersListTemp.add(decoderMapper.toDecoderEntity(decoder.get()));
+      DeviceEntity updatedDevice =
+          maintainDeviceStatus.maintainStatusField(decodersListTemp).get(0);
+      // update the retrieved decoder object
+      decoder.get().getDevice().setStatus(updatedDevice.getStatus());
```

Comment:
should this be in the service

Reply:
We don't have this method in the service yet, though we have a refactoring task in zenhub to move it, so we'll move this code when we do that refactoring task